src/docs/devdoc-schema.dox

```diff
@@ -0,0 +1,116 @@
+/*! @page devdoc-schema Schema operations
+
+A schema defines the format of the application data in WiredTiger. WiredTiger
+supports various types of schemas (See @ref schema for more
+information), operated upon through a WT_SESSION reference. This section details
+the internals of these various schema operations.
+
+Schema operations cause an update to the metadata and are performed under a
+schema lock to avoid concurrent operations on the same object. Following
+sequence of steps define a generic schema operation:
```

Comment:
I think you should sit down with Amar and discuss the content of these two paragraphs - I feel like they could be clearer, but someone less involved with WiredTiger probably has a better perspective.